src/System.Management.Automation/DscSupport/JsonDscClassCache.cs

```diff
@@ -0,0 +1,3274 @@
+                var metaConfigExtensionFile = Path.Combine(systemResourceRoot, "BaseRegistration\\MSFT_MetaConfigurationExtensionClasses.schema.json");
+                ImportClasses(metaConfigExtensionFile, DefaultModuleInfoForMetaConfigResource, errors);
+
+                //
+                // Load all of the system resource schema files, searching
+                //
+                string resources;
+                foreach (var resourceRoot in allResourceRoots)
+                {
+                    resources = Path.Combine(resourceRoot, "Schema");
+                    if (!Directory.Exists(resources))
+                    {
+                        continue;
+                    }
+
+                    foreach (var schemaFile in Directory.EnumerateDirectories(resources).SelectMany(d => Directory.EnumerateFiles(d, "*.schema.json")))
+                    {
+                        ImportClasses(schemaFile, s_defaultModuleInfoForResource, errors);
+                    }
+                }
+
+                // Load Regular and DSC PS modules
+                bool importInBoxResourcesImplicitly = false;
+                List<string> modulePaths = new List<string>();
+                if (modulePathList == null || modulePathList.Count == 0)
+                {
+                    modulePaths.Add(Path.Combine(configSystemPath, inboxModulePath));
+                    importInBoxResourcesImplicitly = true;
+                }
+                else
+                {
+                    foreach (string moduleFolderPath in modulePathList)
+                    {
+                        if (!Directory.Exists(moduleFolderPath))
+                        {
+                            continue;
+                        }
+
+                        foreach (string moduleDir in Directory.EnumerateDirectories(moduleFolderPath))
+                        {
+                            modulePaths.Add(moduleDir);
+                        }
+                    }
+                }
+
+                LoadDSCResourceIntoCache(errors, modulePaths, importInBoxResourcesImplicitly);
+            }
+        }
+
+        /// <summary>
+        /// Load DSC resources into Cache from moduleFolderPath.
+        /// </summary>
+        /// <param name="errors">Collection of any errors encountered during initialization.</param>
+        /// <param name="modulePathList">Module path from where DSC PS modules will be loaded.</param>
+        /// <param name="importInBoxResourcesImplicitly">
+        /// if module is inbox.
+        /// </param>
+        private static void LoadDSCResourceIntoCache(Collection<Exception> errors, List<string> modulePathList, bool importInBoxResourcesImplicitly)
+        {
+            foreach (string moduleDir in modulePathList)
+            {
+                if (!Directory.Exists(moduleDir)) continue;
+
+                var dscResourcesPath = Path.Combine(moduleDir, "DscResources");
+                if (Directory.Exists(dscResourcesPath))
```

Comment:
This is windows only code?

Reply:
Discussed offline. The concern was case-sensitive path comparisons that these APIs are doing on Linux.
The mitigation is that this code is running on Windows only.